packages/ui/src/components/Button/SocialLinkButton.tsx

```diff
@@ -0,0 +1,42 @@
+import { ConnectorMetadata } from '@logto/schemas';
+import classNames from 'classnames';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import * as SocialLinkButtonStyles from './SocialLinkButton.module.scss';
+import * as styles from './index.module.scss';
+
+export type Props = {
+  isDisabled?: boolean;
+  className?: string;
+  connector: Pick<ConnectorMetadata, 'id' | 'name' | 'logo'>;
+  onClick?: (id: string) => void;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const SocialLinkButton = ({ isDisabled, className, connector, onClick = noop }: Props) => {
+  const { id, name, logo } = connector;
+
+  const {
+    i18n: { language },
+  } = useTranslation();
+  const localName = name[language] ?? name.en;
+
+  return (
+    <button
+      disabled={isDisabled}
+      className={classNames(styles.button, SocialLinkButtonStyles.socialButton, className)}
+      type="button"
+      onClick={() => {
+        onClick(id);
+      }}
+    >
+      {logo && <img src={logo} alt={localName} className={SocialLinkButtonStyles.icon} />}
+      {localName}
+    </button>
+  );
+};
+
+export default SocialLinkButton;
```

Comment:
Could this be a type of `Button` component, instead of a separated component itself?

Reply:
Since it requires `connectorMetaData` as input param feels like a little bit distinctive, a separate component might be more clear?

Reply:
I'm good with it, then